integrationtest/termination_recovery_test.go

```
@@ -0,0 +1,160 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	"github.com/pborman/uuid"
+	"github.com/pivotal-cf/brokerapi/v10/domain"
+)
+
+var _ = Describe("Recovery From Broker Termination", func() {
+	const (
+		serviceOfferingGUID = "083f2884-eb7b-11ee-96c7-174e35671015"
+		servicePlanGUID     = "0d953850-eb7b-11ee-bb2c-8ba95d780d82"
+	)
+
+	var (
+		brokerpak string
+		broker    *testdrive.Broker
+		stdout    *Buffer
+		stderr    *Buffer
+	)
+
+	BeforeEach(func() {
+		brokerpak = must(packer.BuildBrokerpak(csb, fixtures("termination-recovery")))
+
+		stdout = NewBuffer()
+		stderr = NewBuffer()
+		broker = must(testdrive.StartBroker(csb, brokerpak, database, testdrive.WithOutputs(stdout, stderr)))
+
+		DeferCleanup(func() {
+			Expect(broker.Stop()).To(Succeed())
+			cleanup(brokerpak)
+		})
+	})
+
+	It("cannot recover from a terminated create", func() {
+		By("starting to provision")
+		instanceGUID := uuid.New()
+		response := broker.Client.Provision(instanceGUID, serviceOfferingGUID, servicePlanGUID, uuid.New(), nil)
+		Expect(response.Error).NotTo(HaveOccurred())
+		Expect(response.StatusCode).To(Equal(http.StatusAccepted))
+
+		By("terminating and restarting the broker")
+		Expect(broker.Stop()).To(Succeed())
+		broker = must(testdrive.StartBroker(csb, brokerpak, database))
+
+		By("INCORRECTLY reporting that an operation is still in progress")
+		lastOperation, err := broker.LastOperation(instanceGUID)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(lastOperation.Description).To(Equal("provision in progress"))
+		Expect(lastOperation.State).To(BeEquivalentTo("in progress"))
+
+		// OSBAPI requires that HTTP 409 (Conflict) is returned
+		By("refusing to allow a duplicate instance")
+		response = broker.Client.Provision(instanceGUID, serviceOfferingGUID, servicePlanGUID, uuid.New(), nil)
+		Expect(response.Error).NotTo(HaveOccurred())
+		Expect(response.StatusCode).To(Equal(http.StatusConflict))
+
```

Out of curiosity, what happens here if we try to update?

---

It fails with an internal server error and:
```
{"timestamp":"1711540247.853883982","source":"cloud-service-broker","message":"cloud-service-broker.unknown-error","log_level":2,"data":{"error":"terraform version check failed: workspace state not generated"}}
```
In a delete we have a special case that ignores this error allowing the cleanup. But we don't do that for update, and failing is probably the right thing to do here.